### DISCUSSION TITLE
How to format the data for base64 value

### DISCUSSION THREAD

**Discussion by drewpo28** [make-all/tuya-local#1511] [General]
Hi all,

I have weather station device that return some data in base64 format, see example below.
`{
        "code": "lcd_night_time",
        "custom_name": "",
        "dp_id": 130,
        "time": 1704940030447,
        "value": "FgAGAA=="
}`

The format of the value is:
Data[0] is the starting clock, 0-23; default 22
Data[1] is the starting minute, 0-59; default 0
Data[2] is the end clock, 0-23; default 6
Data[3] is the end minute, 0-59; default 0

I am able to get the exact byte using mask option, but the question is it possible to create entity where sensor is combination of few values, like Data[0]:Data[1]?

Thanks

**Comment by make-all** [comment]
To combine values back together, you are probably best using a template_sensor, which is very flexible in what you can do with it.

**Reply by drewpo28** [reply]
thank you for answer, is it possible to describe in device.yaml?
I can't find such type of entity :(

**Reply by make-all** [reply]
No, it is not an aim of this project to duplicate the functionality of template sensors.

**Reply by drewpo28** [reply]
I see, thank you a lot
